src/lib/json-parser.c

```diff
@@ -138,7 +138,7 @@ int json_parser_deinit(struct json_parser **_parser, const char **error_r)
 					   i_stream_get_name(parser->input),
 					   i_stream_get_error(parser->input));
 	} else if (parser->data == parser->end &&
-		   !i_stream_have_bytes_left(parser->input) &&
+		   parser->input->eof &&
```

Comment:
Is this eof check needed at all? Looking at the commit ae9691f7ef36d5272d72c90fa51393dfea5dd126 that added i_stream_have_bytes_left() check, previously there wasn't any EOF check in there.